### PR TITLE
Fix many attributes being labeled as worthless if they had another attribute in the name

### PR DESCRIPTION
**Pull request by Nealkitt**
## What
Changes the way bad attributes are checked, I got upset that it was labeling my FS as worthless

## Changelog Fixes
+ Fixed Attributes being labeled as bad if it had another attribute in the name - Stella
    * Now is in a list of attributes

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt [hunk 1]
```diff
@@ -192,15 +192,27 @@ object EstimatedItemValueCalculator {
...
-        return false
+    private val badAttributeList = listOf(
+        // armor
+        "ARACHNO_RESISTANCE",
+        "BLAZING_RESISTANCE",
+        "ENDER_RESISTANCE",
+        "EXPERIENCE",
+        "FORTITUDE",
+        "SPEED",
+        "UNDEAD_RESISTANCE",
```
Comment (lunaynx) [inline src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt:204]: Maybe remove this as it is situationally useful (MP/UR Aurora)?

Comment (Nealkitt) [reply src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt:204]: Yeah, this is temporary until we get a "Good rolls" list added to Repo that I think Empa was working on for a feature

File: src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt [hunk 2]
```diff
@@ -192,15 +193,36 @@ object EstimatedItemValueCalculator {
...
-    private fun isUselessAttribute(internalName: String): Boolean {
-        if (internalName.contains("RESISTANCE")) return true
-        if (internalName.contains("SPEED")) return true
-        if (internalName.contains("EXPERIENCE")) return true
-        if (internalName.contains("FORTITUDE")) return true
-        if (internalName.contains("ENDER")) return true
+    private fun isUselessAttribute(itemName: NEUInternalName, attribute1: String? = null, attribute2: String? = null): Boolean {
+        val list = listOf(attribute1, attribute2)
+        auroraArmorPattern.matchMatcher(itemName.asString()) {
+            return ("MANA_POOL" in list) && ("UNDEAD_RESISTANCE" in list)
```
Comment (lunaynx) [inline src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt]:
```suggestion
            if (("MANA_POOL" in list) && ("UNDEAD_RESISTANCE" in list)) {
                return false
            }
```

File: src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt [hunk 3]
```diff
@@ -192,17 +193,44 @@ object EstimatedItemValueCalculator {
         } else {
             0.0
         }
+    private val badAttributeList = listOf(
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/misc/items/EstimatedItemValueCalculator.kt:196]: please move that list in the repo

**Review by hannibal002 — CHANGES_REQUESTED** [review]
i think there is already another pr/repo stuff planned for detecting god rolls. It should probably be combined, instead of code duplication in the mod and in repo

**Pull request closed without merging**